### PR TITLE
Modified CONTRIBUTING.md

### LINKED ISSUE #2326: Modify the CONTRIBUTING.md file to enhance readability
**Required changes**
- Restructure the doc such that it is much easier to digest
- Provide a detailed guide on environment set up
- Modify grammar
- Fix links
- Fix minor typos

---
**Contributor Resources**

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)
- See Layer5 site designs in this [Figma project](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs). Join the [Layer5 Community](http://slack.layer5.io) for access.

### PR DESCRIPTION
**Pull request by Mr-DG-Wick**
**Description**

This PR fixes #2326

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (shubham-tyagi) [inline CONTRIBUTING.md:144]:
```suggestion
  **12.** Once you’ve committed and pushed all of your changes to GitHub, go to the page for your fork on GitHub, select your development branch, and click the pull request button.. Please ensure that you compare your feature branch to the desired branch of the repo you are supposed to make a PR to.``` If you need to make any adjustments to your pull request, just push the updates to GitHub. Your pull request will automatically track the changes in your development branch and update it.

File: CONTRIBUTING.md
Comment (shubham-tyagi) [inline CONTRIBUTING.md:137]: maybe before pushing the changes, we could ask them to pull changes in the master from upstream and rebase their branch with the master branch. Something like in the below images :

File: CONTRIBUTING.md
Comment (shubham-tyagi) [inline CONTRIBUTING.md:134]: in addition to this command maybe we could give them an option for a relatively shorter command `git commit -s -m "<commit subject>"`

**Review by shubham-tyagi — CHANGES_REQUESTED** [review]
Hey @Mr-DG-Wick, Nice structuring, Great stuff. But I am not sure whether `package-lock.json` should be modified or not. Maybe @alphaX86 or @warunicorn19 can help us to confirm this.

File: CONTRIBUTING.md
Comment (shubham-tyagi) [inline CONTRIBUTING.md:138]: maybe before pushing the changes, we could ask them to pull changes in the master from upstream and rebase their branch with the master branch. Something like in the below images :

File: CONTRIBUTING.md
Comment (alphaX86) [inline CONTRIBUTING.md:30]:
```suggestion
- First step is to set up the local development environment. See [this](#How to contribute) on how to do the same. 
- Bug fixes are always welcome. Start by reviewing the [list of bugs](https://github.com/layer5io/layer5/labels/kind%2Fbug).
- A good way to easily start contributing is to pick and work on a [good first issue](https://github.com/layer5io/layer5/labels/good%20first%20issue). We try to make these issues as clear as possible and provide basic info on how the code should be changed, and if something is unclear feel free to ask for more information on the issue.
- We regularly discuss about new issues to work on in our [#websites](https://layer5io.slack.com/archives/C015QJKUMPU) channel on slack. Feel free to join and discuss about any issue or any idea that you may have.
```

File: CONTRIBUTING.md
Comment (alphaX86) [inline CONTRIBUTING.md:15]: You left the most important part of all xD

```suggestion
## Newcomers guide
To help you get started on contributing to the Layer5 projects, we made a newcomers guide to make contributions easier for new folks like you! Refer the [newcomers guide](https://layer5.io/community/newcomers) for the resources and tutorials to help you get started!

## Style guide
```

File: CONTRIBUTING.md
Comment (alphaX86) [inline CONTRIBUTING.md:166]:
```suggestion
  $ git commit -s -m “my commit message w/signoff”
```

**Review by alphaX86 — CHANGES_REQUESTED** [review]
Do the following changes @Mr-DG-Wick

**Comment by alphaX86** [comment]
> Hey @Mr-DG-Wick, Nice structuring, Great stuff. But I am not sure whether `package-lock.json` should be modified or not. Maybe @alphaX86 or @warunicorn19 can help us to confirm this.

It's best not to modify `package-lock.json` file as it may break CI/CD. Just saw this

**Comment by warunicorn19** [comment]
Yeah, so only when you make UI changes or changes on the actual website then you need to commit the `package-lock.json` but not while making changes to the Readme

**Comment by alphaX86** [comment]
@Mr-DG-Wick restore `package-lock.json`

**Comment by l5io** [comment]
🚀 Preview for commit 11e30900b71cad1a36633060014ab71d5e2ba559 at: https://6187b1d3689b2901ea7d4724--layer5ng.netlify.app

**Comment by Mr-DG-Wick** [comment]
Thank you for reviewing @shubham-tyagi @alphaX86 @warunicorn19 ! I shall commit suggested change asap :)

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:15]: I missed it xd

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:134]: Yup for sure!

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:144]: Nice point!

**Comment by Mr-DG-Wick** [comment]
I don't why or how `package-lock.json` file got modified but I'll revert the changes back. I had various issues with WSL and git so maybe something went wrong in there.

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:30]: @alphaX86  Did you suggest that I indent it with tabs? If yes then, I did it now :)

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:19]: done!

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:138]: done!

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md]: done!

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md]: done!

**Comment by Mr-DG-Wick** [comment]
> @Mr-DG-Wick restore `package-lock.json` file

Donee

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:146]: sounds good.

**Comment by alphaX86** [comment]
image

Fix this one @Mr-DG-Wick and we're good to go

File: CONTRIBUTING.md
Comment (alphaX86) [inline CONTRIBUTING.md:222]:
```suggestion
- <img src="https://github.com/layer5io/layer5/blob/master/src/sections/Community/Member-single/getnighthawk-logo.png" width="25px" height="25px"/> Nighthawk
```

**Comment by alphaX86** [comment]
That's all other than that good

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [inline CONTRIBUTING.md:177]: @alphaX86 @shubham-tyagi @warunicorn19 Can anyone of you please share how you set up your .gitconfig file for the commits cuz I'm having some issues with DCO so thought I'll add these too in the doc so that it is much easier for others too on how to set up commits with the DCO sign up.
I checked some of your commits in other PR's and when you compare it with my commits they look different.

Comment (alphaX86) [reply CONTRIBUTING.md:177]: In .gitconfig huh... let me check...

Comment (alphaX86) [reply CONTRIBUTING.md:177]: image

This is mine actually and I didn't add any of the things xD Just enter -s that's all

Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:177]: so you mean :
`$ git commit -s -m “New Change”`
That's it?
Cuz I used to do this : 
`$ git commit --signoff -m " Commited suggestions_2  Signed-off-by: Deepak Gupta <deepakgdkg1g8868@gmail.com>"`
and weirdly it doesn't look like how you all commit. It looks like this :
Screenshot (1082)

Comment (alphaX86) [reply CONTRIBUTING.md:177]: > Cuz I used to do this :
$ git commit --signoff -m " Commited suggestions_2 Signed-off-by: Deepak Gupta <deepakgdkg1g8868@gmail.com>"

Well you don't need to include that Sign-off line in commit message actually :') Git automatically makes one if you give -s or --signoff flag

Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:177]: So @alphaX86, finally simply do this right?
`$ git commit -s -m “New Change”`

this confused me:
`$ commit -s -m “my commit message w/signoff”`

Comment (shubham-tyagi) [reply CONTRIBUTING.md:177]: Yes, I too just add -s in my git commit command. We don't need to include signoff in the actual commit message.
`git commit -s -m "<commit msg>"` is good enough.

Comment (Mr-DG-Wick) [reply CONTRIBUTING.md:177]: Aight thanks @alphaX86 @shubham-tyagi .

File: CONTRIBUTING.md
Comment (Mr-DG-Wick) [reply CONTRIBUTING.md]: Donee

**Comment by Mr-DG-Wick** [comment]
> Fix this one @Mr-DG-Wick and we're good to go

Done!

**Comment by Mr-DG-Wick** [comment]
What do ya'll think about the new navigation part added at the top? @alphaX86 @shubham-tyagi @warunicorn19 ?

**Comment by alphaX86** [comment]
> What do ya'll think about the new navigation part added at the top? @alphaX86 @shubham-tyagi @warunicorn19 ?

👍 🔥 🔥  Superb @Mr-DG-Wick

**Comment by shubham-tyagi** [comment]
> What do ya'll think about the new navigation part added at the top? @alphaX86 @shubham-tyagi @warunicorn19 ?

Awesome 👌👏

**Review by warunicorn19 — APPROVED** [review]
LGTM 🎉, Thanks @Mr-DG-Wick, let's wait for @leecalcote approval.

**Review by alphaX86 — APPROVED** [review]
LGTM as well 🥳

**Comment by l5io** [comment]
🚀 Preview for commit 92c462dfec18202fb032e18cb001545dc1154e39 at: https://61890cd30b8dbb0071938214--layer5ng.netlify.app

**Comment by shubham-tyagi** [comment]
LGTM as well, nice work @Mr-DG-Wick  🎉

**Review by leecalcote — APPROVED** [review]
Looks great!

**Pull request merged**
